File: sdks/java/lib/src/main/java/models/shared/SignRequestSignerInput.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * SignRequestSignerInput
 * Prefill tags are used to prefill placeholders with signer input data. Only
 * one value field can be included.
**/
public class SignRequestSignerInput {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("checkbox_value")
    public Boolean checkboxValue;
    public SignRequestSignerInput withCheckboxValue(Boolean checkboxValue) {
        this.checkboxValue = checkboxValue;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("content_type")
    public SignRequestSignerInputContentTypeEnum contentType;
    public SignRequestSignerInput withContentType(SignRequestSignerInputContentTypeEnum contentType) {
        this.contentType = contentType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("date_value")
    public OffsetDateTime dateValue;
    public SignRequestSignerInput withDateValue(OffsetDateTime dateValue) {
        this.dateValue = dateValue;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("document_tag_id")
    public String documentTagId;
    public SignRequestSignerInput withDocumentTagId(String documentTagId) {
        this.documentTagId = documentTagId;
        return this;
    }
    @JsonProperty("page_index")
    public Long pageIndex;
    public SignRequestSignerInput withPageIndex(Long pageIndex) {
        this.pageIndex = pageIndex;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("text_value")
    public String textValue;
    public SignRequestSignerInput withTextValue(String textValue) {
        this.textValue = textValue;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public SignRequestSignerInputTypeEnum type;
    public SignRequestSignerInput withType(SignRequestSignerInputTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: go-sdk/pkg/models/shared/signrequestsignerinput.go
package shared

import (
	"time"
)

type SignRequestSignerInputContentTypeEnum string

const (
	SignRequestSignerInputContentTypeEnumInitial   SignRequestSignerInputContentTypeEnum = "initial"
	SignRequestSignerInputContentTypeEnumStamp     SignRequestSignerInputContentTypeEnum = "stamp"
	SignRequestSignerInputContentTypeEnumSignature SignRequestSignerInputContentTypeEnum = "signature"
	SignRequestSignerInputContentTypeEnumCompany   SignRequestSignerInputContentTypeEnum = "company"
	SignRequestSignerInputContentTypeEnumTitle     SignRequestSignerInputContentTypeEnum = "title"
	SignRequestSignerInputContentTypeEnumEmail     SignRequestSignerInputContentTypeEnum = "email"
	SignRequestSignerInputContentTypeEnumFullName  SignRequestSignerInputContentTypeEnum = "full_name"
	SignRequestSignerInputContentTypeEnumFirstName SignRequestSignerInputContentTypeEnum = "first_name"
	SignRequestSignerInputContentTypeEnumLastName  SignRequestSignerInputContentTypeEnum = "last_name"
	SignRequestSignerInputContentTypeEnumText      SignRequestSignerInputContentTypeEnum = "text"
	SignRequestSignerInputContentTypeEnumDate      SignRequestSignerInputContentTypeEnum = "date"
	SignRequestSignerInputContentTypeEnumCheckbox  SignRequestSignerInputContentTypeEnum = "checkbox"
)

type SignRequestSignerInputTypeEnum string

const (
	SignRequestSignerInputTypeEnumSignature SignRequestSignerInputTypeEnum = "signature"
	SignRequestSignerInputTypeEnumDate      SignRequestSignerInputTypeEnum = "date"
	SignRequestSignerInputTypeEnumText      SignRequestSignerInputTypeEnum = "text"
	SignRequestSignerInputTypeEnumCheckbox  SignRequestSignerInputTypeEnum = "checkbox"
)

// SignRequestSignerInput
// Prefill tags are used to prefill placeholders with signer input data. Only
// one value field can be included.
type SignRequestSignerInput struct {
	CheckboxValue *bool                                  `json:"checkbox_value,omitempty"`
	ContentType   *SignRequestSignerInputContentTypeEnum `json:"content_type,omitempty"`
	DateValue     *time.Time                             `json:"date_value,omitempty"`
	DocumentTagID *string                                `json:"document_tag_id,omitempty"`
	PageIndex     int64                                  `json:"page_index"`
	TextValue     *string                                `json:"text_value,omitempty"`
	Type          *SignRequestSignerInputTypeEnum        `json:"type,omitempty"`
}
